protocol/nats_jetstream/v3/protocol.go

```diff
@@ -0,0 +1,100 @@
+/*
+ Copyright 2021 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package nats_jetstream
+
+import (
+	"context"
+
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/protocol"
+
+	"github.com/nats-io/nats.go"
+)
+
+// Protocol is a reference implementation for using the CloudEvents binding
+// integration. Protocol acts as both a NATS client and a NATS handler.
+type Protocol struct {
+	conn *nats.Conn
+
+	Consumer *consumer
+
+	Sender *sender
```

Comment:
Regarding the naming, should it be `consumer-producer` or `sender-receiver` to better align with conventions? @stephen-totty-hpe @embano1

Reply:
+1 e.g., sender/receiver

Reply:
Removed Consumer.  Everything is folded into Receiver